splunk_add_on_ucc_framework/install_python_libraries.py

```diff
@@ -152,3 +162,65 @@ def remove_execute_bit(installation_path: str) -> None:
             logger.info(f"  fixing {o} execute bit")
             current_permissions = stat.S_IMODE(os.lstat(o).st_mode)
             os.chmod(o, current_permissions & no_exec)
+
+
+def install_os_dependent_libraries(
+    package_path: str, ucc_lib_target: str, installer: str
+) -> None:
+    conf_file_name = "os-dependent_packages.json"
+    config_file = os.path.join(package_path, "lib", conf_file_name)
+
+    if not os.path.isfile(config_file):
+        logger.info(
+            "Config file not found. Installation of os-dependent libraries skipped."
+        )
+        return
+
+    try:
+        with open(config_file) as file:
+            packages = json.load(file)["libraries"]
```

Comment:
Is it ok to pull pydantic for types/formats? Named tuples are cool, but it may be a more standard approach to deal with schemas and serialization.
Having a standard process in defining UCC configuration formats should be a priority as TAs depend on it.
Having a json schema to be referred by a TA addon?
Can this be an extension of a `globalConfig.json` for consistency instead of a separate file? Having a versioned schema for `globalConfig.json` would be nice too.

Reply:
That's what I was thinking about regarding pedantic or anything else which gives us types from the globalConfig. We just recently finished refactoring it so it can be used to generate types, frontend folks are already working on that.

Regarding schema for globalConfig - we have it in splunk_add_on_ucc_framework/schema/schema.json file.

Regarding pulling this config into globalConfig file - I did not want to bring more responsibility to globalConfig, but probably it would be more consistent and easier to use.